### DISCUSSION TITLE
Does LLamaSharp expose llama.cpp's finetuning functionality?

### DISCUSSION THREAD

**Discussion by ErectableSentry** [SciSharp/LLamaSharp#706] [Q&A]
I can see that I can apply LoRA adapters to a model via LLamaSharp, but just curious about whether I can generate those adapters via this library (and if not, any suggestions on achieving this within my existing .net project?)

As shown in llama.cpp's example here: https://github.com/ggerganov/llama.cpp/tree/master/examples/finetune

**Comment by SanftMonster** [comment]
Finetune has not been supported by LLamaSharp yet. Could you please tell us something about how would you like to integrate the finetune functionality with your app? For example, handling input data with C#, finetune in the desktop app, or anything else?

**Reply by ErectableSentry** [reply]
Thanks, yea I'm looking to create a high-level standard training pipeline for a single model. My input data is of the same format, so my thinking is that once configured & verified, I won't have a great need to touch the lower levels of the training pipeline, simply pass my data to my c# app, it does it's thing, and outputs the adapter. 

I suppose for this requirement I can bypass LLamaSharp and work with llama.cpp directly, but that'll increase the project complexity.

**Comment by SanftMonster** [comment]
Finetune is in our roadmap, but at this time we're in short of hand to do it. May I ask in which scenario your training pipeline will work? There're lots of python libraries providing functionalities to set up LLM training within one step. So it seems that you want to put the training and inference together in an app? I'll appreciate for your answer because that helps us to see how important this feature is and how to introduce it in the future. :)
